operations/fetch.go

```diff
@@ -518,7 +518,7 @@ type artifactDownload struct {
 
 func getArtifactFolderName(task *service.RestTask) string {
 	if evergreen.IsPatchRequester(task.Requester) {
-		return fmt.Sprintf("artifacts-patch-%v_%v_%v", task.PatchNumber, task.BuildVariant, task.DisplayName)
```

Comment:
if the build variant is less than 100 characters this will panic; we should case on this. Also shouldn't we do this in the non-patch-requester case as well?

Reply:
I think it would be worth adding a unit test for this function, to ensure the output does what you want.